lua/core/info.lua

```diff
@@ -130,13 +130,26 @@ end
 
 function M.toggle_popup(ft)
   local client = u.get_active_client_by_ft(ft)
-  local is_client_active = not client.is_stopped()
-  local client_enabled_caps = require("lsp").get_ls_capabilities(client.id)
+  local is_client_active = false
+  local client_enabled_caps = {}
+  local client_name = ""
+  local client_id = 0
+  local document_formatting = false
+  local missing_linters = {}
+  local missing_formatters = {}
   local num_caps = vim.tbl_count(client_enabled_caps)
   local null_ls_providers = null_ls_handler.get_registered_providers_by_filetype(ft)
-
-  local missing_linters = lvim.lang[ft].linters._failed_requests or {}
-  local missing_formatters = lvim.lang[ft].formatters._failed_requests or {}
+  if client ~= nil then
+    is_client_active = not client.is_stopped()
+    client_enabled_caps = require("lsp").get_ls_capabilities(client.id)
```

Comment:
Should the `num_caps` be calculated again after getting them?

Reply:
yes they should you are right